src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx

```diff
@@ -21,69 +21,134 @@ type Props = {
-                min={min}
-                max={max}
-                dataFilter={dataFilter}
-              >
-                {results => (
-                  <React.Fragment>
-                    {vitals.map((vital, index) => {
-                      const error =
-                        summaryResults.error !== null || results.error !== null;
-                      const alias = getAggregateAlias(
-                        `percentile(${vital}, ${PERCENTILE})`
-                      );
-                      const summary =
-                        summaryResults.tableData?.data?.[0]?.[alias] ?? null;
-                      return (
-                        <VitalCard
-                          key={vital}
-                          location={location}
-                          isLoading={summaryResults.isLoading || results.isLoading}
-                          error={error}
-                          vital={WEB_VITAL_DETAILS[vital]}
-                          summary={summary as number | null}
-                          chartData={results.histograms?.[vital] ?? []}
-                          colors={[colors[index]]}
-                          eventView={eventView}
-                          organization={organization}
-                          min={min}
-                          max={max}
-                        />
-                      );
-                    })}
+            <React.Fragment>
+              {vitals.map((vital, index) => {
+                const alias = getAggregateAlias(`percentile(${vital}, ${PERCENTILE})`);
+                const summary = summaryResults.tableData?.data?.[0]?.[alias] ?? null;
+                const vitalSlug = WEB_VITAL_DETAILS[vital].slug;
+
+                return (
+                  <React.Fragment key={vital}>
+                    {this.renderVitalCard(
+                      vital,
+                      isLoading,
+                      error,
+                      summary,
+                      multiHistogramResults.histograms?.[vital] ?? [],
+                      [colors[index]],
+                      decodeScalar(location.query[`${vitalSlug}Start`]),
+                      decodeScalar(location.query[`${vitalSlug}End`])
+                    )}
                   </React.Fragment>
-                )}
-              </MeasurementsHistogramQuery>
-            </Panel>
+                );
+              })}
+            </React.Fragment>
           );
         }}
-      </DiscoverQuery>
+      </MeasurementsHistogramQuery>
+    );
+  }
+
+  render() {
+    const {location, organization, eventView} = this.props;
+
+    return (
+      <Panel>
+        <DiscoverQuery
+          location={location}
+          orgSlug={organization.slug}
+          eventView={eventView}
+          limit={1}
+        >
+          {results => (
+            <React.Fragment>
+              {VITAL_GROUPS.map((vitalGroup, i) => {
+                return (
+                  <React.Fragment key={i}>
```

Comment:
Does the sort order ever change on these components? Using the index offset as a key won't work if the data can be resorted on subsequent renders. Instead you'd want to combine the index with something like a name or slug that is more closely related to the object's identity.

Reply:
I choose the index as the key here because the order is predefined and is not expected to change. I can compose the slugs together as the key if that's better.

Reply:
If the order doesn't change you should be ok. If you notice stale rendering states it could be caused by the key.